athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/domain/spill/SpillLocationVerifier.java

```diff
@@ -0,0 +1,115 @@
+package com.amazonaws.athena.connector.lambda.domain.spill;
+
+/*-
+ * #%L
+ * Amazon Athena Query Federation SDK
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class is used to track the bucket and its state, and check its validity
+ */
+public class SpillLocationVerifier
+{
+    private static final Logger logger = LoggerFactory.getLogger(SpillLocationVerifier.class);
+
+    private enum BucketState
+    {UNCHECKED, VALID, INVALID}
+
+    private final AmazonS3 s3;
+    private String bucket;
+    private BucketState state;
+
+    /**
+     * @param bucket The name of the spill bucket.
+     */
+    public SpillLocationVerifier(AmazonS3 s3, String bucket)
```

Comment:
Why are you requiring a bucket in the constructor? Sorry if I missed this in the last revision but it seems needless you should only use the bucket provided to checkBucketAuthZ(...)

Reply:
The SpillLocationVerifier is used to verify the validity of the spill bucket, so if there's no spill bucket, there's no need to initialize a SpillLocationVerifier. Instead of assigning an empty string to "bucket" in SpillLocationVerifier, the user of this class should know what the bucket is. If the user doesn't know the bucket at initialization, they should consciously pass an empty string to the constructor of SpillLocationVerifier. IMO.

Reply:
You dont need the spill bucket in the constructor to do this. For example, when you create an S3 client you dont need to tell it at construction time what S3 key you plan on reading instead you do that when you call getObject. The same applies here, since the bucket you are being asked to check can change there is little point to taking it as a constructor argument.

Reply:
This makes sense. I'll remove the bucket from the constructor.